db/migrate/20240917092519_create_custom_option_hierarchies.rb

```diff
@@ -0,0 +1,16 @@
+class CreateCustomOptionHierarchies < ActiveRecord::Migration[7.1]
+  def change
+    create_table :custom_option_hierarchies, id: false do |t|
```

Comment:
🟠 I would leave the id present- ActiveRecord won't play well with nil primary key. For example, `#destroy` won't work which means you can't leverage things like AR callbacks.

Of course you have the option to [define composite primary keys](https://guides.rubyonrails.org/active_record_composite_primary_keys.html) but I've recently learnt [_the hard way_](https://github.com/opf/openproject/pull/16694) the change in data structure can be messy